src/webots/gui/WbSingleTaskApplication.cpp

```diff
@@ -44,13 +48,91 @@ void WbSingleTaskApplication::run() {
   else if (mTask == WbGuiApplication::VERSION)
     cout << tr("Webots version: %1").arg(WbApplicationInfo::version().toString(true, false, true)).toUtf8().constData() << endl;
   else if (mTask == WbGuiApplication::UPDATE_PROTO_CACHE)
-    updateProtoCacheFiles(mTaskArgument);
+    updateProtoCacheFiles();
   else if (mTask == WbGuiApplication::UPDATE_WORLD)
     WbWorld::instance()->save();
+  else if (mTask == WbGuiApplication::CONVERT) {
+    convertProto();
+  }
 
   emit finished(mTask == WbGuiApplication::FAILURE ? EXIT_FAILURE : EXIT_SUCCESS);
 }
 
+void WbSingleTaskApplication::convertProto() const {
+  QCommandLineParser cliParser;
+  cliParser.setApplicationDescription("Convert a PROTO file to URDF, WBO, or WRL file");
+  cliParser.addHelpOption();
+  cliParser.addPositionalArgument("input", "Path to the input PROTO file.");
+  cliParser.addPositionalArgument("output", "Path to the output URDF, WBO, or WRL file.");
+  cliParser.addOption(QCommandLineOption("p", "Override default PROTO parameters.", "parameter=value"));
+  cliParser.process(mTaskArguments);
+  QStringList positionalArguments = cliParser.positionalArguments();
+  if (positionalArguments.size() != 2)
+    cliParser.showHelp(1);
+
+  // Compute absolute paths for input and output files
+  QString inputFile = positionalArguments[0];
+  QString outputFile = positionalArguments[1];
+  if (QDir::isRelativePath(inputFile))
+    inputFile = mStartupPath + '/' + inputFile;
+  if (QDir::isRelativePath(outputFile))
+    outputFile = mStartupPath + '/' + outputFile;
+
+  // Get user parameters strings
+  QMap<QString, QString> userParameters;
+  for (QString param : cliParser.values("p")) {
+    QStringList pair = param.split("=");
+    if (pair.size() != 2) {
+      cout << tr("A parameter is not properly formated!\n").toUtf8().constData();
+      cliParser.showHelp(1);
+    }
+    userParameters[pair[0]] = pair[1].replace(QRegExp("^\"*"), "").replace(QRegExp("\"*$"), "");
+  }
+
+  // Parse PROTO
+  new WbProtoList(QFileInfo(inputFile).absoluteDir().path());
+  WbNode::setInstantiateMode(false);
+  WbProtoModel *model = WbProtoList::current()->readModel(inputFile, "");
+  cout << tr("Parsing the %1 PROTO...").arg(model->name()).toUtf8().constData() << endl;
+
+  // Combine the user parameters with the default ones
+  QVector<WbField *> fields;
+  for (WbFieldModel *model : model->fieldModels()) {
+    WbField *field = new WbField(model);
+    if (userParameters.contains(field->name())) {
+      WbTokenizer tokenizer;
+      tokenizer.tokenizeString(userParameters[field->name()]);
+      field->readValue(&tokenizer, "");
+    }
+
+    cout << tr("  field %1 [%2] = %3")
+              .arg(field->name())
+              .arg(field->stringType())
```

Comment:
This is more consistent for users:
```suggestion
              .arg(field->value()->vrmlTypeName())
```
and we could remove the stringType() method.